src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/QualityGuidelines/CSharpAddMissingInterpolationToken.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpAddMissingInterpolationToken : AbstractAddMissingInterpolationTokenAnalyzer
+    {
+        private protected override bool ShouldReport(ILiteralOperation operation)
+        {
+            var annotation = new SyntaxAnnotation();
+            if (SyntaxFactory.ParseExpression("$" + operation.Syntax.ToString(), options: operation.Syntax.SyntaxTree.Options).WithAdditionalAnnotations(annotation)
+                is not InterpolatedStringExpressionSyntax dummyNode)
+            {
+                return false;
+            }
+
+            var root = operation.Syntax.SyntaxTree.GetRoot();
+            root = root.ReplaceNode(operation.Syntax, dummyNode);
+            dummyNode = (InterpolatedStringExpressionSyntax)root.GetAnnotatedNodes(annotation).Single();
```

Comment:
I am not sure if that's a good solution to throw in case of multiple matches.

Reply:
@Evangelink This should never happen. We annotated one node only. So `GetAnnotatedNodes` shouldn't return multiple nodes.